### ISSUE TITLE
Support for ENV variable in freemarker templates

### ISSUE THREAD

**Issue by tdipisa** [geosolutions-it/geoserver#200]
It would be useful to have a support for the ENV variable in GeoServer Freemarker templates. For example in GetFeatureInfo requests, where the locale sent by the client as a query parameter in the URL could be used.

http://mydomain/geoserver/wms?service=WMS&version=1.1.1&request=GetFeatureInfo&exceptions=application/json&id=layer_name_propr__3&layers=layer_name&query_layers=layer_name&x=51&y=51&height=101&width=101&srs=EPSG:3857&bbox=1262445.155988286,5859892.496590748,1262475.3127846017,5859922.653387064&feature_count=10&info_format=text/html**&ENV=mapstore_language:it**

Having this support available in GeoServer, the corresponding part in FTL template could be:

**Comment by nmco** [comment]
Client and project:
```
AUSTROCONTROL
C125-2020-AUSTROCONTROL-Map2Imp
```

@fernandor777 looks like this should already be supported as per [this documentation,](https://docs.geoserver.org/latest/en/user/tutorials/freemarker.html#examples) specially this example: ``${request.ENV.PROPERTY}``.

Can you double check if this is support, and if yes what's the syntax? Otherwise, please prepare a plan to implement it and estimate it.

**Comment by fernandor777** [comment]
An update about ENV variables support on GetFeatureInfo Freemarker templates:

Using the ENV definition on the URL:
```
&ENV=mapstore_language:it_IT
```
The variable usage in the template should be:
```
${request.ENV.MAPSTORE_LANGUAGE}
```

Example content template definition (data/templates/content.ftl):
```
<#assign locale="${request.ENV.MAPSTORE_LANGUAGE}">
<#setting locale="${request.ENV.MAPSTORE_LANGUAGE}">

<p>${request.ENV.MAPSTORE_LANGUAGE}</p>
<p>${locale}</p>

<ul>
<#list features as feature>
  <li><b>Type: ${type.name}</b> (id: <em>${feature.fid}</em>):
  <ul>
  <#list feature.attributes as attribute>
    <#if !attribute.isGeometry>
      <li>${attribute.name}: ${attribute.value}</li>
    </#if>
  </#list>
  </ul>
  </li>
</#list>
</ul>
```

Example output:
![Screenshot from 2020-07-27 10-23-23](https://user-images.githubusercontent.com/3613119/88560552-e6899400-cff3-11ea-859c-e64771cb88bb.png)

**Comment by nmco** [comment]
@tdipisa let's know if you need more details.

**Comment by tdipisa** [comment]
@fernandor777 @nmco ok thank you. therefore it is already supported. Did I understand correctly?

**Comment by nmco** [comment]
Yes, that say @fernandor777 when you have 5 minutes can you do a PR adding a more meaningful example to the documentation?

**Comment by tdipisa** [comment]
@nmco @fernandor777 one more question please: is this available also on 2.16.2?

**Comment by fernandor777** [comment]
@tdipisa yes, I tested it with GeoServer 2.16.2 and it is supported.

**Comment by tdipisa** [comment]
@fernandor777 ok thank you so much.
